tests/std/tests/GH_004109_iter_value_t_direct_initialization/test.compile.pass.cpp

```diff
@@ -0,0 +1,700 @@
+
+    // Set operations (on sorted ranges)
+    ranges::merge(nil, nil, nil, nil, nil);
+    ranges::merge(nil, nil, nil, nil, nil, comp);
+
+    ranges::inplace_merge(nil, nil, nil);
+    ranges::inplace_merge(nil, nil, nil, comp);
+
+    (void) ranges::includes(nil, nil, nil, nil);
+    (void) ranges::includes(nil, nil, nil, nil, comp);
+
+    ranges::set_difference(nil, nil, nil, nil, nil);
+    ranges::set_difference(nil, nil, nil, nil, nil, comp);
+
+    ranges::set_intersection(nil, nil, nil, nil, nil);
+    ranges::set_intersection(nil, nil, nil, nil, nil, comp);
+
+    ranges::set_symmetric_difference(nil, nil, nil, nil, nil);
+    ranges::set_symmetric_difference(nil, nil, nil, nil, nil, comp);
+
+    ranges::set_union(nil, nil, nil, nil, nil);
+    ranges::set_union(nil, nil, nil, nil, nil, comp);
+
+    // Heap operations
+    (void) ranges::is_heap(nil, nil);
+    (void) ranges::is_heap(nil, nil, comp);
+
+    (void) ranges::is_heap_until(nil, nil);
+    (void) ranges::is_heap_until(nil, nil, comp);
+
+    ranges::make_heap(nil, nil);
+    ranges::make_heap(nil, nil, comp);
+
+    ranges::push_heap(nil, nil);
+    ranges::push_heap(nil, nil, comp);
+
+    ranges::pop_heap(nil, nil);
+    ranges::pop_heap(nil, nil, comp);
+
+    ranges::sort_heap(nil, nil);
+    ranges::sort_heap(nil, nil, comp);
+
+    // Minimum/maximum operations
+    (void) ranges::max_element(nil, nil);
+    (void) ranges::max_element(nil, nil, comp);
+
+    (void) ranges::min_element(nil, nil);
+    (void) ranges::min_element(nil, nil, comp);
+
+    //(void) ranges::minmax_element(nil, nil);
+    //(void) ranges::minmax_element(nil, nil, comp);
+
+    // Permutation operations
+    (void) ranges::is_permutation(nil, nil, nil, nil);
+    (void) ranges::is_permutation(nil, nil, nil, nil, pred2);
+
+    ranges::next_permutation(nil, nil);
+    ranges::next_permutation(nil, nil, comp);
+
+    ranges::prev_permutation(nil, nil);
+    ranges::prev_permutation(nil, nil, comp);
+
+    // Constrained numeric operations
+#if _HAS_CXX23
+    ranges::iota(nil, nil, val);
+#endif // _HAS_CXX23
+
+    // Constrained fold operations
+#if _HAS_CXX23
+    (void) ranges::fold_left(nil, nil, val, binop);
+    (void) ranges::fold_left_first(nil, nil, binop);
+    (void) ranges::fold_right(nil, nil, val, binop);
+    (void) ranges::fold_right_last(nil, nil, binop);
+    (void) ranges::fold_left_with_iter(nil, nil, val, binop);
+    (void) ranges::fold_left_first_with_iter(nil, nil, binop);
+#endif // _HAS_CXX23
+
+    // Constrained uninitialized memory algorithms
+    ranges::uninitialized_copy(nil, nil, nil, nil);
+    ranges::uninitialized_copy_n(nil, zero, nil, nil);
+
+    ranges::uninitialized_fill(nil, nil, val);
+    ranges::uninitialized_fill_n(nil, zero, val);
+
+    ranges::uninitialized_move(nil, nil, nil, nil);
+    ranges::uninitialized_move_n(nil, zero, nil, nil);
+
+    ranges::destroy(nil, nil);
+
+    ranges::destroy_n(nil, zero);

```

Comment:
`destroy_at` and `construct_at` are not bound to iterator mechanisms. `uninitialized_default_construct` should be unrelated because it should use default-initialization only.

It seems to me that `destroy` and `destroy_n` are also unrelated as they don't initialize any element.